### ISSUE TITLE
Support ARM-based M1 Macs to run Amundsen in Docker

### ISSUE THREAD

**Issue by cmpadden** [amundsen-io/amundsen#1812]
## Expected Behavior or Use Case

It should be possible to run Amundsen locally on M1 Mac computers. Currently, errors are thrown regarding the ARM-based architecture.

## Possible Implementation

We could make an ARM-specific `docker-compose.yml` file, or we could conditionally load _arm64_ images depending on the platform of the host machine:

```
if [[ $(uname -m) == 'arm64' && $(uname) == 'Darwin' ]]; then 
    ...
fi
```

I was able to get `docker-amundsen.yml` to work with the following change:

```diff
@@ -2,6 +2,7 @@ version: '3'
 services:
   neo4j:
       image: neo4j:3.5.26
+      platform: linux/amd64
       container_name: neo4j_amundsen
       environment:
         - NEO4J_AUTH=neo4j/test
```

## Context

```
> docker-compose -f docker-amundsen.yml up
Pulling neo4j (neo4j:3.5.26)...
3.5.26: Pulling from library/neo4j
ERROR: no matching manifest for linux/arm64/v8 in the manifest list entries
```

```
> docker-compose -f docker-amundsen-atlas.yml up
...
Pulling elasticsearch (elasticsearch:6.8.13)...
6.8.13: Pulling from library/elasticsearch
ERROR: no matching manifest for linux/arm64/v8 in the manifest list entries
```

```
> uname -a
Darwin cpadden-m01 21.4.0 Darwin Kernel Version 21.4.0: Fri Mar 18 00:46:32 PDT 2022; root:xnu-8020.101.4~15/RELEASE_ARM64_T6000 arm64
```

_EDIT:_ additionally, it appears there may be a compatibility issue with the _fsevents_ dependency in `docker-amundsen-atlas.yml`.

```
#15 64.90 Build failed with error code: 1
#15 66.00 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/chokidar/node_modules/fsevents):
#15 66.00 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"arm64"})
#15 66.00 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.3 (node_modules/fsevents):
#15 66.00 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"arm64"})
#15 66.00
#15 66.01 npm ERR! code ELIFECYCLE
#15 66.02 npm ERR! errno 1
#15 66.02 npm ERR! node-sass@5.0.0 postinstall: `node scripts/build.js`
#15 66.02 npm ERR! Exit status 1
#15 66.02 npm ERR!
#15 66.02 npm ERR! Failed at the node-sass@5.0.0 postinstall script.
```

**Comment by randopop** [comment]
just noticed this was closed. i assume this is not a priority?

**Comment by cmpadden** [comment]
> just noticed this was closed. i assume this is not a priority?

I've not heard back from anyone on this issue, so that may be the case.

**Comment by yoku2010** [comment]
I was also facing the same issue with M1 Mac.

Screenshot of error:
<img width="1006" alt="arm-issue" src="https://user-images.githubusercontent.com/1465530/180656774-132ac9f5-7798-401e-a966-5f1ab4603f94.png">

But after adding `platform: linux/amd64` in `docker-compose.yml` file, it got fixed.

Then I faced another issue:
> Error response from daemon: Ports are not available: exposing port TCP 0.0.0.0:5000 -> 0.0.0.0:0: listen tcp 0.0.0.0:5000: bind: address already in use

Screenshot of error:
<img width="1067" alt="Screenshot 2022-07-24 at 9 58 31 PM" src="https://user-images.githubusercontent.com/1465530/180657576-834c9133-0f6d-4ee0-aa8d-ff833e6e1b79.png">

This was apparently due to the new AirPlay functionality. Control Center stops listening to those ports when I turned off “AirPlay Receiver” in the “Sharing” System Preference.

Screenshot of solution:
<img width="664" alt="Screenshot 2022-07-24 at 10 08 41 PM" src="https://user-images.githubusercontent.com/1465530/180657631-519535f6-381a-4ac6-a641-d882a9b08acb.png">

Now everything is workingfine.

__I am wondering, why apple chose such a popular port.__
